pkg/util/common.go

```diff
@@ -199,3 +199,92 @@ func TagStringToSemver(tag string) (semver.Version, error) {
 func SemverToTagString(tag semver.Version) string {
 	return AddTagPrefix(tag.String())
 }
+
+// CopyFile copies a file from one location to another.
+func CopyFile(src, dst string, ignoreNotExist bool) error {
```

Comment:
I'm wondering if we should be more explicit here around how we're copying.
From what I've seen/understood about the code, there are at least two copy operations:
- Locally staging the artifacts
- Copy from local staging to GCS

Maybe use `CopyLocal[File|Dir]` and `CopyToGCS`?